### PR TITLE
LIve Editor Component - Dynamic Routing File Hiearchy

### PR DESCRIPTION
**Pull request by samzhao273**
**Added Live Editor Component to Cosmic Gateway Repo**

**What:** Added live editor component for docs and implement dynamic routing for the docs folder. Added dependencies for tiptap/svelte for component. Integrated editor with dynaimc file hiearchy.

**Why/How**:  Installed dependencies, used example markdown text as placeholder for the doc fetch from db. To ensure routing for the project still works.

**Ticket:**https://github.com/ubclaunchpad/CosmicGateway/issues/45

### CODE REVIEW

File: src/routes/(home)/doc/doc1Group/[id]/+page.svelte
Comment (armintalaie) [inline src/routes/(home)/doc/doc1Group/[id]/+page.svelte:1]: should delete this file! + the doc1group folder

File: package.json [hunk 1]
```diff
@@ -36,6 +37,16 @@
 	},
 	"type": "module",
 	"dependencies": {
-		"jwt-decode": "^4.0.0"
```
Comment (armintalaie) [inline package.json:39]: this shouldnt be deleted

**Review by armintalaie — CHANGES_REQUESTED** [review]
some minor changes, rerequest a review when u have it updates.

other than that great work

File: package.json (hunk 1 above)
Comment (samzhao273) [reply package.json:39]: Its just moved below the other extensions - will move it back to the original line

**Review by armintalaie — APPROVED** [review]

**Pull request merged**